composer.json

```diff
@@ -33,7 +33,7 @@
         "dflydev/fig-cookies": "^2.0.1 || ^3.0",
         "laminas/laminas-cli": "^0.1.5 || ^1.0",
         "laminas/laminas-diactoros": "^1.8 || ^2.0",
-        "laminas/laminas-httphandlerrunner": "^1.0.1",
+        "laminas/laminas-httphandlerrunner": "^1.0.1 || ^2.1",
```

Comment:
Didn't check the rest of the patch yet, but can we just bump to `^2.1` and drop `^1.0.1` instead please?

Reply:
No, we can't. It would break BC for anybody extending the `SwooleRequestHandlerRunner` currently. We'd have to bump major to make it happen.

I'm only doing it here because for some reason, we bumped the minimum supported in mezzio/mezzio, so right now you cannot install mezzio-swoole against a new skeleton install. (We shouldn't have bumped in mezzio/mezzio without a major version change; the difference in what `Mezzio\Application` typehints against in the constructor changed entirely, which is what is causing breakage with alternate laminas-httphandlerrunner implementations.)

I'm happy to do an immediate v4 release after this, but I think we need it in the v3 series first.

Reply:
Actually, been thinking on this overnight, and I think I'll just target a new major version with this change, and release 3.6 immediately with the other changes I've merged this week. WIll reduce maintenance.

Reply:
Done now, @Ocramius . 95% of this PR is just copying documentation for v4, so that I can add a migration guide.